client/driver/docker.go

```diff
@@ -1115,6 +1132,32 @@ func (d *DockerDriver) createContainerConfig(ctx *ExecContext, task *structs.Tas
 	}
 	hostConfig.Privileged = driverConfig.Privileged
 
+	// set capabilities
+	hostCapsWhitelist := d.config.ReadDefault(dockerCapsWhitelistConfigOption, dockerCapsWhitelistConfigDefault)
+	hostCapsWhitelist = strings.ToLower(hostCapsWhitelist)
+	if !strings.Contains(hostCapsWhitelist, "all") {
+		effectiveCaps, err := caps.TweakCapabilities(
+			strings.Split(dockerBasicCaps, ","),
+			driverConfig.CapAdd,
+			driverConfig.CapDrop,
+		)
+		if err != nil {
+			return c, err
+		}
+		var missingCaps []string
+		for _, cap := range effectiveCaps {
+			cap = strings.ToLower(cap[len("CAP_"):])
+			if !strings.Contains(hostCapsWhitelist, cap) {
```

Comment:
This logic is susceptible to imprecise substring matches. Despite no existing capability being a substring of another capability, I still feel like this risks a future exploit by allowing "admin" to match "net_admin" in the whitelist.

Can we either parse the cap whitelist into a map to do key lookups or surround the whitelist with delimiters (`,`) and search for the delimited form: `,net_admin,`.

I know capabilities don't change often (ever anymore?), but I'd still rather be safe than sorry when doing any sort of whitelist lookups.

Reply:
Oh yeah, absolutely :) This is just a quick poc to get some feedback. I'll try finish this over the weekend.

Reply:
Great! Yeah, your general approach looks great.